pajbot/modules/casechecker.py

```diff
@@ -37,55 +37,160 @@ class CaseCheckerModule(BaseModule):
+            type="number",
+            required=True,
+            placeholder="",
+            default=8,
+            constraints={"min_value": 0, "max_value": 500},
+        ),
+        ModuleSetting(
+            key="lowercase_percentage",
+            label="Maximum percent of lowercase letters allowed in message",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 0, "max_value": 100},
+        ),
+        ModuleSetting(
+            key="max_lowercase",
+            label="Maximum amount of lowercase characters allowed in a message",
+            type="number",
+            required=True,
+            placeholder="",
+            default=50,
+            constraints={"min_value": 0, "max_value": 500},
+        ),
+        ModuleSetting(
+            key="uppercase_timeouts", label="Enable uppercase timeouts", type="boolean", required=True, default=False
+        ),
+        ModuleSetting(
+            key="uppercase_timeout_duration",
+            label="Uppercase Timeout duration",
+            type="number",
+            required=True,
+            placeholder="",
+            default=3,
+            constraints={"min_value": 3, "max_value": 120},
         ),
         ModuleSetting(
             key="uppercase_timeout_reason",
             label="Uppercase Timeout Reason",
             type="text",
             required=False,
             placeholder="",
-            default="no uppercase characters allowed",
-            constraints={},
+            default="Too many uppercase characters",
+            constraints={"max_str_len": 500},
         ),
         ModuleSetting(
-            key="lowercase_timeout_reason",
-            label="Lowercase Timeout Reason",
-            type="text",
-            required=False,
+            key="min_uppercase_characters",
+            label="Minimum amount of uppercase characters before checking for a percentage",
+            type="number",
+            required=True,
             placeholder="",
-            default="NO LOWERCASE CHARACTERS ALLOWED",
-            constraints={},
+            default=8,
+            constraints={"min_value": 0, "max_value": 500},
+        ),
+        ModuleSetting(
+            key="uppercase_percentage",
+            label="Maximum percent of uppercase letters allowed in message",
+            type="number",
+            required=True,
+            placeholder="",
+            default=60,
+            constraints={"min_value": 0, "max_value": 100},
+        ),
+        ModuleSetting(
+            key="max_uppercase",
+            label="Maximum amount of uppercase characters allowed in a message",
+            type="number",
+            required=True,
+            placeholder="",
+            default=50,
+            constraints={"min_value": 0, "max_value": 500},
         ),
```

Comment:
These settings should change order to fit with the functionality better.
Right now, the `max_uppercase` and `max_lowercase` settings are the ones that take priority, yet they're at the bottom of the configuration.
It might also be worth adding to the `max_uppercase/lowercase` descriptions that they are the values that are checked first